### PR TITLE
Sort vehicles correctly in build vehicle window

### LINKED ISSUE #1393: vehicle list not in order when playing savegames or scenario
Version information:
 - Windows 10
 - 22.03.1
 - don't know what build/commit it is on.

The vehicle list is not in order that they were invented in the build trains window. For an example, a GP38-2 that was built in 1972 is after a P42DC on the list that was built in 1996. the gp38 should be in between a J Class and a C40-8W

To Reproduce:
Steps to reproduce the behaviour:
1. Go to a savegame or scenario
2. Click on 'build trains'
3. look the list of trains and the year that they were invented
4. you will see that some trains that are invented a year or two that is after another train that was invented a year or two later

Expected behaviour:
The list in the window should be on order the year they were invented.

### PR DESCRIPTION
**Pull request by LeftofZen**
Fixes #1393

### CODE REVIEW

File: src/OpenLoco/Windows/BuildVehicle.cpp [hunk 1]
```diff
@@ -482,13 +483,13 @@ namespace OpenLoco::Ui::Windows::BuildVehicle
                 }
             }
 
-            auto power = std::min<uint16_t>(vehicleObj->power, 1);
-            // Unsure why power is only checked for first byte.
-            buildableVehicles.push_back({ vehicleObjIndex, static_cast<uint8_t>(power), vehicleObj->designed });
+            // 'power' is a flag indicating if the vehicle is powered or not. 1 = powered, 0 = unpowered
+            auto power = static_cast<uint8_t>(std::min<uint16_t>(vehicleObj->power, 1));
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/BuildVehicle.cpp:487]: should we change this to a bool? `isPowered`? Tbh i think it should just compare the whole of the power value may as well order it completely by power.

Comment (LeftofZen) [reply src/OpenLoco/Windows/BuildVehicle.cpp:487]: `isPowered` is a good idea, I'll implement that tomorrow.

I'm just thinking about the 'final version' of the build vehicles window, and I can see one obvious set of enhancements is to add the table-sorting column headers, like OpenTTD does for its 'build vehicles' window:
image

With those changes, there wouldn't even be a need to sort by date-then-power as you'd be able to filter out vehicles by type and then sort by an individual statistic.

Comment (AaronVanGeffen) [reply src/OpenLoco/Windows/BuildVehicle.cpp:487]: Hmm. I'm a little apprehensive about adding table/sort header buttons to lists that don't have obvious columns, to be honest. I think the OpenTTD screenshot isn't very intuitive UI.

Comment (LeftofZen) [reply src/OpenLoco/Windows/BuildVehicle.cpp:487]: Sure, I'm not saying copy the OpenTTD UI, I'm saying copy the functionality.
Being able to only display vehicles that are powered, or vehicles that carry passengers, etc, would be an immensely useful feature in my opinion.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**